python/pyarrow/types.py

```diff
@@ -64,17 +82,17 @@ def is_boolean(t):
 
 @doc(is_null, datatype="any integer")
 def is_integer(t):
-    return t.id in _INTEGER_TYPES
+    return is_integer_type(t)
 
 
 @doc(is_null, datatype="signed integer")
 def is_signed_integer(t):
-    return t.id in _SIGNED_INTEGER_TYPES
+    return is_signed_integer_type(t)
 
 
 @doc(is_null, datatype="unsigned integer")
 def is_unsigned_integer(t):
-    return t.id in _UNSIGNED_INTEGER_TYPES
+    return is_unsigned_integer_type(t)
 
 
 @doc(is_null, datatype="int8")
```

Comment:
There are no functions exposed on a one-to-one basis in `type_traits.h` for these types.

Reply:
I would suggest checking for `bit_width(t)` together with `is_integer_type(t)`. Both are exposed in this PR.